### PR TITLE
STORM-1348 - refactor API to remove Insert/Update builder in Cassandra connector

### PR DESCRIPTION
**Pull request by fhussonnois**
I have to add some tests but I open this PR to start its review.

### CODE REVIEW

File: external/storm-cassandra/src/main/java/org/apache/storm/cassandra/DynamicStatementBuilder.java [hunk 1]
```diff
@@ -19,10 +19,11 @@
 package org.apache.storm.cassandra;
 
 import com.datastax.driver.core.BatchStatement;
+import com.datastax.driver.core.querybuilder.BuiltStatement;
 import org.apache.storm.cassandra.query.*;
```
Comment (satishd) [inline external/storm-cassandra/src/main/java/org/apache/storm/cassandra/DynamicStatementBuilder.java]: Remove wild card imports.

File: external/storm-cassandra/src/main/java/org/apache/storm/cassandra/query/builder/BoundCQLStatementMapperBuilder.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.storm.cassandra.query.builder;
+
+import org.apache.storm.cassandra.query.*;
```
Comment (satishd) [inline external/storm-cassandra/src/main/java/org/apache/storm/cassandra/query/builder/BoundCQLStatementMapperBuilder.java]: Remove wild card imports.

File: external/storm-cassandra/src/main/java/org/apache/storm/cassandra/query/builder/BoundCQLStatementMapperBuilder.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import org.apache.storm.cassandra.query.impl.BoundCQLStatementTupleMapper;
+import org.apache.storm.cassandra.query.impl.PreparedStatementBinder;
+import org.apache.storm.cassandra.query.impl.RoutingKeyGenerator;
+import org.apache.storm.cassandra.query.selector.FieldSelector;
+
+import java.io.Serializable;
+import java.util.*;
```
Comment (satishd) [inline external/storm-cassandra/src/main/java/org/apache/storm/cassandra/query/builder/BoundCQLStatementMapperBuilder.java]: Remove wild card imports.

File: external/storm-cassandra/src/main/java/org/apache/storm/cassandra/query/builder/BoundCQLStatementMapperBuilder.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+import static org.apache.storm.cassandra.query.ContextQuery.*;
```
Comment (satishd) [inline external/storm-cassandra/src/main/java/org/apache/storm/cassandra/query/builder/BoundCQLStatementMapperBuilder.java]: Remove wild card imports.

File: external/storm-cassandra/src/main/java/org/apache/storm/cassandra/query/builder/SimpleCQLStatementMapperBuilder.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.storm.cassandra.query.builder;
+
+import com.datastax.driver.core.querybuilder.BuiltStatement;
+import org.apache.storm.cassandra.query.*;
```
Comment (satishd) [inline external/storm-cassandra/src/main/java/org/apache/storm/cassandra/query/builder/SimpleCQLStatementMapperBuilder.java]: Remove wild card imports.

File: external/storm-cassandra/src/main/java/org/apache/storm/cassandra/query/impl/PreparedStatementBinder.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+import com.datastax.driver.core.*;
+import org.apache.storm.cassandra.query.Column;
+
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.net.InetAddress;
+import java.nio.ByteBuffer;
+import java.util.*;
```
Comment (satishd) [inline external/storm-cassandra/src/main/java/org/apache/storm/cassandra/query/impl/PreparedStatementBinder.java]: Remove wild card imports.

File: external/storm-cassandra/src/test/java/org/apache/storm/cassandra/DynamicStatementBuilderTest.java [hunk 7]
```diff
@@ -56,78 +61,93 @@
         when(mockTuple.getFields()).thenReturn(new Fields("weatherstation_id", "event_time", "temperature"));
     }
 
+    public static final String QUERY_STRING = "INSERT INTO weather.temperature(weatherstation_id,event_time,temperature) VALUES (?,?,?);";
+
```
Comment (satishd) [inline external/storm-cassandra/src/test/java/org/apache/storm/cassandra/DynamicStatementBuilderTest.java]: Change weatherstation_id to weather_station_id which is aligned with actual name in the table mentioned in resources.

File: external/storm-cassandra/src/test/java/org/apache/storm/cassandra/DynamicStatementBuilderTest.java [hunk 8]
```diff
@@ -56,78 +61,93 @@
         when(mockTuple.getFields()).thenReturn(new Fields("weatherstation_id", "event_time", "temperature"));
```
Comment (satishd) [inline external/storm-cassandra/src/test/java/org/apache/storm/cassandra/DynamicStatementBuilderTest.java]: "weatherstation_id", "event_time", "temperature" can be declared as constants and used across the class.

**Comment by satishd** [comment]
@fhussonnois Can you merge with latest master and address remaining review comments?

**Comment by fhussonnois** [comment]
@satishd done ;)

**Comment by satishd** [comment]
+1

**Comment by fhussonnois** [comment]
@harshach is this PR can be merge if modifications are OK for you ? Thank you.

**Comment by ptgoetz** [comment]
+1

**Comment by revans2** [comment]
+1

**Pull request merged**